lib/rouge/lexer.rb

```diff
@@ -501,12 +521,47 @@ def self.detect?(text)
   end
 
   module Lexers
+    ProxyLexer = Struct.new :path, :name, :tag, :aliases, :filenames, :mimetypes
+
+    @_loaded_files = {}
     @_loaded_lexers = {}
+    @_loaded_proxies = {}
+    @_source_files = []
+
+    def self.add_lexer(class_name)
+      return if @_source_files.empty?
+      @_loaded_files[class_name] = @_source_files.last
+    end
 
     def self.load_lexer(relpath)
       return if @_loaded_lexers.key?(relpath)
       @_loaded_lexers[relpath] = true
+      @_source_files.push relpath
       load File.join(__dir__, 'lexers', relpath)
+      @_source_files.pop
+    end
+
+    def self.load_proxies(hashes)
+      hashes.each do |h|
+        next if h["tag"].nil?
+        proxy = ProxyLexer.new(h["path"], h["name"].to_sym, h["tag"],
+                               h["filenames"], h["mimetypes"])
+        @_loaded_proxies[proxy.name] = proxy
+      end
+
+      registry = {}
+      @_loaded_proxies.values.each do |proxy|
+        registry[proxy.tag] = proxy
+        proxy.aliases.each { |a| registry[a] = proxy } unless proxy.aliases.nil?
```

Comment:
```suggestion
        proxy.aliases.each { |a| registry[a] = proxy } if proxy.aliases
```

But it would be better if `proxy.aliases` was stashed in a local var:
```ruby
if (proxy_aliases = proxy.aliases)
  proxy_aliases.each { |a| ... }
end
```

Reply:
Would it? We'd lose the conciseness of this being a one-liner and it's inconsistent with the other uses around it of accessing properties of the Struct with the dot.